Upgrade to walletconnect 2.0.0-rc.3

h0ngcha0 opened the pull request.

h0ngcha0 committed:
Upgrade dependencies

h0ngcha0 committed:
Improve the namespaces handling logic

h0ngcha0 committed:
Verify provider.chains

h0ngcha0 committed:
Upgrade packages

h0ngcha0 committed:
Add comments about the relay image

polarker commented on src/index.ts:
```diff
@@ -41,6 +42,8 @@ export const signerMethods = [
 type SignerMethodsTuple = typeof signerMethods;
 type SignerMethods = SignerMethodsTuple[number];
 
+export type ChainGroup = -1 | 0 | 1 | 2 | 3
```
I tend to avoid the assumption of 4 groups in our tools whenever it's possible, as we will increase the group number and build L2 chains.

Type side, `ChainGroup = number | undefined` might be less confusing than `-1`. `undefined` for devs and `-1` for the underlying protocol.

There is also an inconsistency as far as I see. Sometimes the code uses `[alephium:4/0, alephium:4/1, alephium:4/2, alephium:4/3]`, sometime the code uses `alephium:4/-1`

h0ngcha0 replied:
Sounds great, `undefined` feels a lot more dev friendlier than `-1` :+1:

polarker commented on src/index.ts:
```diff
@@ -74,39 +77,56 @@ interface SignerMethodsTable extends Record<SignerMethods, { params: any; result
+  chainGroup: ChainGroup;
+}
+
+export const ALEPHIUM_NAMESPACE = "alephium"
```
👍

polarker commented on src/index.ts:
```diff
@@ -290,52 +318,56 @@ class WalletConnectProvider implements SignerProvider {
     address: address,
     publicKey: publicKey,
-    group: groupOfAddress(address),
+    group: +group
```
My initial thought was that we should trust as less data as possible, that's why the group is calculated from address.

But I should have only relayed public key and calculate both address and public key from it. WDYT?

h0ngcha0 replied:
Yeah, that is a good point, lets do that :+1:

polarker reviewed (commented):
Just done a quick review. A lot of nice improvement, great work 👍

h0ngcha0 committed:
Update ChainGroup data type

h0ngcha0 committed:
Relay publicKey instead of both publicKey & address

h0ngcha0 committed:
Fix lint and test in CI

h0ngcha0 committed:
Fix test in CI

h0ngcha0 committed:
Fix wrong error message in CI

h0ngcha0 committed:
Minor cleanup

h0ngcha0 committed:
Add mergePermittedChainsInfo

h0ngcha0 committed:
Use -1 to represet all groups in chain id

h0ngcha0 committed:
Test getPermittedChainId

h0ngcha0 committed:
Test signing for all groups

polarker committed:
Minor refactorization

This PR tries to make the WC provider consistent the following design:

- WC provider could initialize a session with a namespace defining the permitted network and chain group, only one network would be allowed.
- WC client (desktop wallet, mobile wallet) has to approve the namespace and select an address for the group. Depending on the permitted namespace, WC client's active address and network may be switched.
- WC provider can not change the address. It can change the network, but the respective WC session will be closed immediately.
- WC clients can change the active address. All such changes would trigger session_update events if the address is in the right group. Otherwise, the session will be closed.
- WC clients may change the network, but the active session will be closed immediately.

polarker committed:
Fix the type of chainGroup configuration

h0ngcha0 committed:
Fix all tests

h0ngcha0 committed:
Test getPermittedChainGroups

h0ngcha0 committed:
Add more signer methods

h0ngcha0 committed:
Test/fix signer methods

polarker committed:
Minor updates

h0ngcha0 committed:
Only allow single permitted network with single chain group (number | -1)

polarker committed:
Merge pull request #8 from alephium/minor-refactorization

Minor refactorization
